src/devices/Hx711I2c/Hx711I2c.cs

```diff
@@ -0,0 +1,349 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Buffers.Binary;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Device.Model;
+using System.Diagnostics;
+using System.Threading;
+using UnitsNet;
+
+namespace Iot.Device.Hx711
+{
+    /// <summary>
+    /// DFRobot KIT0176: I2C 1kg Weight Sensor Kit
+    /// </summary>
+    [Interface("DFRobot KIT0176: I2C 1kg Weight Sensor Kit")]
+    public class Hx711I2c : IDisposable
+    {
+        /// <summary>
+        /// Default address for I2C, only use when pins A0 and A1 are set to 0.
+        /// Otherwise use GetI2cAddress
+        /// </summary>
+        public const int DefaultI2cAddress = 0x64;
+
+        /// <summary>
+        /// Arbitrarily picked value for empty scale.
+        /// </summary>
+        public const float DefaultOffset = 6780606.5f;
+
+        private I2cDevice _i2c;
+
+        /// <summary>
+        /// Raw value telling where 0 is.
+        /// It will be set to current weight when Tare function is used.
+        /// Value passed must be a raw reading.
+        /// </summary>
+        [Property]
+        public float Offset { get; set; }
+
+        /// <summary>
+        /// Value which scales raw units into grams
+        /// </summary>
+        [Property]
+        public float CalibrationScale { get; set; }
+
+        /// <summary>
+        /// When set to true, CAL button will not have any effect on the current calibration setting
+        /// </summary>
+        [Property]
+        public bool IgnoreCalibrationButton { get; set; }
+
+        /// <summary>
+        /// When set to true, RST button will not change Offset (it won't Tare)
+        /// </summary>
+        [Property]
+        public bool IgnoreResetButton { get; set; }
+
+        /// <summary>
+        /// Sets the weight (in grams) used for automatic calibration.
+        /// Value is only relevant when CAL button has been pressed.
+        /// </summary>
+        [Property]
+        public ushort AutomaticCalibrationWeight
+        {
+            // It does not seem to be possible to read from this register
```

Comment:
even if not possible, you can store the data you write?
Same for the other ones?

Reply:
I actually don't initially write anything for these two but I suppose I could initialize them in the ctor